internal/kernel_SSE.h

```diff
@@ -0,0 +1,332 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// kernel_SSE.h: a collection of Intel SSE optimized kernels.
+// Check in kernel_default.h which one(s) are actually used by default.
+// Others are mere experiments; they are still covered by tests
+// in case they might be useful some day.
+// Writen by: Sagi Marovich, mail: sagi.marcovich@intel.com
+
+#ifndef GEMMLOWP_INTERNAL_KERNEL_SSE_H_
+#define GEMMLOWP_INTERNAL_KERNEL_SSE_H_
+
+
+#include "kernel.h"
+#include <cassert>
+#include <string.h>
+#include <iostream>
+
+
+
+
+namespace gemmlowp {
+
+#ifdef GEMMLOWP_SSE32
+struct SSE32Kernel4x4Depth2 : KernelBase {
```

Comment:
Hmm, 4x4 is really small. I understand that with depth 2, with only 8 xmm regs, that's all you can do. So maybe depth 2 is not suitable here, and instead you could do a 8x4 kernel with depth 1. That's what Eigen does.